build-utils/css/constants/utils.ts

```diff
@@ -0,0 +1,24 @@
+type Invalid<T extends string = string> = {
+  message: T;
+};
+
+export const constraintAminoVar = <
+  T extends Record<`--amino-${string}`, string>
+>(object: {
+  [P in keyof T]: T[P];
+}) => object;
+
+export const constraintDefinedAminoVar = <
+  U extends object,
+  T extends Record<string, string> = Record<string, string>
+>(
+  _match: U,
+  object: Exclude<
+    {
+      [P in keyof T]: P extends keyof U
+        ? T[P]
+        : Invalid<`Key need to match keys of the object passed in the first argument`>;
+    },
+    Invalid
+  >
+) => object;
```

Comment:
This is the script contains functions to constraint key of object you put in and produce a object with literal key/value that is accessible for typescript

Reply:
Its awesome! It would be cool if it could run every time a variable was added

Reply:
This is a useless function that will do nothing at runtime since it just returns the same thing. The only time this function would be helpful is when you are developing, it will let you get a union of literal key/value from the object and constraint the key in the object with a specific rule.